airbyte-integrations/bases/source-acceptance-tests/README.md

```diff
@@ -0,0 +1,55 @@
+1. Provide a low overhead way to test basic functionality for any Airbyte source. 
+1. Require minimum effort from the author of the integration.
+
+### What this library _does not_ do 
+1. Test corner cases for specific integrations.
+1. Replace good unit testing and thorough integration testing for the integration.
+
+In order to guarantee a baseline of quality for its users, Airbyte will _not_ merge sources that cannot pass these tests unless the author can provide a very good reason™.
+
+## What's provided by this library?
+This library is structured as follows: 
+1. **Test Banks**: Each test bank verifies that a particular category of connector behaves correctly. We currently provide the following banks: 
+    1. **Core**: Tests basic functionality that every Airbyte source must fulfill
+    1. **Full Refresh**: Any source which supports full refresh syncs on any of its streams should pass this test suite
+    1. **Incremental**: Any source which supports incremental syncs on any of its streams should pass this test suite 
+1. **Test Runner**: This is the entry point which invokes the library. Currently, the test runner accepts a YAML configuration which declares which test suites should run and provides the appropriate inputs to them
+
+
+## Cool. I'm in. How can I get started?
+
+### TL;DR 
+1. Configure your tests via `acceptance_tests.yaml`
+2. Create files needed for tests (e.g: Configured Catalog, connector configuration, etc..)
+3. Invoke the test library via Docker, mounting both your `acceptance_tests.yaml` and any files needed for the tests (the ones you created in step 2) into the Docker image
+
+### In more details
+If you generated your source using one of Airbyte's template generators, you should already have a YAML file in your connector module root called `acceptance_tests.yaml`. If not, create one.
+
+`acceptance_tests.yml` should adhere to the schema described by this [YAML file](src/main/resources/schemas/source_acceptance_test_inputs.yaml). See the linked YAML schema for a description of what each field means.
+
+Once the YAML configuration is ready, from your connector module root simply run `docker run -v $(pwd):/test_inputs/ -w /test_inputs/ airbyte/source-acceptance-tests`. This will mount your entire connector module directory into the test runner's filesystem, thereby making available any files you referenced in `acceptance_tests.yaml`. 
+
+That's it. That's the whole thing. 
+
+If your tests fail, follow the information provided by the test output to identify the problems. 
+
+### FAQ
+**I wrote my connector in language X. Can I use this test suite?**
+
+Yes! As long as your connector is packaged in a Docker container, this library can test it. 
+
+**I need to spin up and tear down resources before/after my tests. How can I do this if this library only accepts YAML?**
+
+We recommend: 
+1. Packaging your test resources in a Docker image that you build before invoking this test library and then
```

Comment:
typo - `this test library.`?

thinking out loud: spinning up resources in a separate docker should be okay since devs shouldn't need to perform complex state mutation for the test. ie.g. I'd expect all users need to do is create the connector resource e.g. Postgres etc and populate it with a couple of records for the test to pass. does that sound right?

Reply:
yup that's the idea. Complex state mutation is not straightforward to do especially polyglot.